libs/mixer/melody.ts

```diff
@@ -110,6 +110,65 @@ namespace music {
 
     let playToneID = 0
 
+    /**
+     * Play a melody from the melody editor.
+     * @param melody - string of up to eight notes [C D E F G A B C5] separated by spaces, 
+     * which will be played one at a time, eg: "E D G F B A C5 B "
+     * @param tempo - number in beats per minute (bpm), dictating how long each note will play for
+     */
+    //% block="play melody $melody at tempo $tempo|(bpm)" blockId=playMelody
+    //% blockNamespace=music
+    //% weight=85 blockGap=8
+    //% group="Melody"
+    //% melody.shadow="melody_editor"
+    //% tempo.min=40 tempo.max=500
+    //% tempo.defl=120
+    export function playMelody(melody: string, tempo: number) {
+        let notes: string[] = melody.split(" ");
```

Comment:
This will be a bit more resilient to whitespace if you filter out the empty strings
```suggestion
        let notes: string[] = melody.split(" ").filter(n => !!n);
```